### DISCUSSION TITLE
What to do about "window is not defined" error when instantiating client in server-side code?

### DISCUSSION THREAD

**Discussion by Stokestack** [supabase/supabase#28099] [Questions]
Hi all. I have a server written using Deno, and I've just switched it over to use Supabase. But when I try to instantiate the Supabase client in my code,

`export const supabase = createClient<Database>('http://localhost:54321', 'eyJhbGciOiJblahblahblahMblYTn_I0')`

it throws an exception from GoTrueClient.js:
```
Exception has occurred: ReferenceError: window is not defined
  at SupabaseAuthClient._isImplicitGrantFlow 
```
[This bug report](https://github.com/supabase/supabase-js/issues/786) seems to be similar, but the provided workaround appears to be React-specific. I'm very new to Supabase, and new to the rest of the stack. So I have no idea how to solve this.

Any insight appreciated!

**Comment by j4w8n** [comment]
Issue filed [here](https://github.com/supabase/supabase-js/issues/1249), and seems to be resolved.

**Reply by Stokestack** [reply]
Thanks. That's the one I filed, but I don't think I'd call it resolved. There is a workaround, but the problem still occurs if you use the API as documented.
